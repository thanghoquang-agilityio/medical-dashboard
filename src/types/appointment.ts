import {
  APIRelatedResponse,
  APIResponse,
  MetaResponse,
  STATUS_TYPE_RESPONSE,
  UserModel,
} from '@/types';

export interface AppointmentModel {
  startTime: string;
  durationTime: string;
  status: keyof typeof STATUS_TYPE_RESPONSE;
  senderId: APIRelatedResponse<APIResponse<UserModel>>;
  receiverId: APIRelatedResponse<APIResponse<UserModel>>;
}

export type AppointmentResponse = APIResponse<AppointmentModel>;

export type AppointmentsDataResponse = Promise<
  { appointments: AppointmentResponse[]; error?: Error } & MetaResponse
>;

export type AppointmentsResponse = {
  data: AppointmentResponse[];
  meta: MetaResponse;
};

export type AppointmentPayload = Omit<
  AppointmentModel,
  'senderId' | 'receiverId'
> & {
  senderId: number;
  receiverId: number;
};

export type AppointMentFormData = Omit<AppointmentPayload, 'status'> & {
  startDate: string;
  status: string;
};
