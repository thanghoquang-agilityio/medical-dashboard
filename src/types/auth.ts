import { UserModel } from './user';

export interface LoginFormData {
  identifier: string;
  password: string;
  remember: boolean;
}

export interface SignupFormData {
  username: string;
  email: string;
  password: string;
  confirmPassWord: string;
}

export type AuthResponse = {
  jwt: string;
  user: UserModel | null;
  error: string | null;
};
