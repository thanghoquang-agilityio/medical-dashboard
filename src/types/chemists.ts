import {
  APIRelatedResponse,
  APIResponse,
  MetaResponse,
  UserModel,
} from '@/types';

export type ChemistModel = {
  users_permissions_user: APIRelatedResponse<APIResponse<UserModel>>;
};

export type ChemistsDataResponse = Promise<
  { chemists: ChemistResponse[]; error: string | null } & MetaResponse
>;

export type ChemistsResponse = {
  data: ChemistResponse[];
  meta: MetaResponse;
};

export type ChemistPayload = {
  users_permissions_user: string;
};

export type ChemistResponse = {
  error: string | null;
};
