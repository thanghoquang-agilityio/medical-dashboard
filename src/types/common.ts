// Components
export type CustomClassType = {
  customClass?: string;
};

export type PageErrorProps = {
  error: Error & { digest?: string };
  reset: () => void;
};

// Status
export enum STATUS_TYPE {
  SUCCESS = 'success',
  WARNING = 'warning',
  ERROR = 'error',
}

// Theme mode
export enum THEME_MODE_TYPE {
  DARK = 'dark',
  LIGHT = 'light',
}

export const STATUS_TYPE_RESPONSE = {
  0: STATUS_TYPE.SUCCESS,
  1: STATUS_TYPE.WARNING,
  2: STATUS_TYPE.ERROR,
} as const;

// Option
export interface Option {
  key: string;
  label: string;
}
