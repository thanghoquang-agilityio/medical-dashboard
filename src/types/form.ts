export interface SignInForm {
  email: string;
  password: string;
}

export interface SignUpFormValue extends SignInForm {
  confirmPassWord: string;
}
