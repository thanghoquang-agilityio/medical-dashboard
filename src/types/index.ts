export * from './common';
export * from './components';
export * from './searchParams';
export * from './status';
