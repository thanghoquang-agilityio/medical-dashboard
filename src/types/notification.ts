import {
  APIRelatedResponse,
  APIResponse,
  MetaResponse,
  UserModel,
} from '@/types';

export interface NotificationModel {
  senderName: string;
  senderAvatar: string;
  isRead: boolean;
  info: JSON;
  senderId?: APIRelatedResponse<APIResponse<UserModel>>;
}

export type NotificationResponse = APIResponse<NotificationModel>;

export type NotificationsDataResponse = Promise<
  { Notifications: NotificationResponse[]; error?: Error } & MetaResponse
>;

export type NotificationsResponse = {
  data: NotificationResponse[];
  meta: MetaResponse;
};

export type NotificationPayload = Omit<NotificationModel, 'senderId'> & {
  senderId: number;
};
