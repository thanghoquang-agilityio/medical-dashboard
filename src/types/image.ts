import { APIResponse } from '@/types';

export type ImageExtension = 'webp' | 'png';

type Formats = {
  large?: { url: string };
  small?: { url: string };
  medium?: { url: string };
  thumbnail?: { url: string };
};

export interface ImageModel {
  name: string;
  hash?: string;
  ext?: ImageExtension;
  formats?: Formats;
  url: string;
}

export type ImageResponse = APIResponse<ImageModel>;
