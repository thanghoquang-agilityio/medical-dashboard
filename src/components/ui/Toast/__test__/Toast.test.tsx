import { render, screen, fireEvent } from '@testing-library/react';

// Types
import { STATUS_TYPE } from '@/types';

// Components
import { Toast } from '..';

describe('Toast component', () => {
  const mockOnClose = jest.fn();

  it('renders the success toast correctly', () => {
    const { container } = render(
      <Toast
        message="Item moved successfully."
        status={STATUS_TYPE.SUCCESS}
        onClose={mockOnClose}
      />,
    );

    expect(screen.getByText('Item moved successfully.')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('renders the error toast correctly', () => {
    render(
      <Toast
        message="Item has been deleted."
        status={STATUS_TYPE.ERROR}
        onClose={mockOnClose}
      />,
    );

    expect(screen.getByText('Item has been deleted.')).toBeInTheDocument();

    expect(screen.getByTestId('toast-icon')).toHaveClass('text-danger-100');
  });

  it('renders the warning toast correctly', () => {
    render(
      <Toast
        message="Improve password difficulty."
        status={STATUS_TYPE.WARNING}
        onClose={mockOnClose}
      />,
    );

    expect(
      screen.getByText('Improve password difficulty.'),
    ).toBeInTheDocument();

    expect(screen.getByTestId('toast-icon')).toHaveClass('text-warning');
  });

  it('calls onClose when close button is clicked', () => {
    render(
      <Toast
        message="Item moved successfully."
        status={STATUS_TYPE.SUCCESS}
        onClose={mockOnClose}
      />,
    );

    const closeButton = screen.getByRole('button');
    fireEvent.click(closeButton);

    expect(mockOnClose).toHaveBeenCalledTimes(1);
  });
});
