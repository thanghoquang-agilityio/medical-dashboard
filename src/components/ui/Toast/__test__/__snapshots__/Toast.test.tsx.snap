// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast component renders the success toast correctly 1`] = `
<div>
  <div
    class="overflow-hidden h-auto text-foreground box-border outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium transition-transform-background motion-reduce:transition-none flex flex-row fixed z-[999] items-center justify-between w-full max-w-xs p-4 rounded-lg top-5 right-5 bg-background-100"
    data-testid="toast"
    role="alert"
    tabindex="-1"
  >
    <svg
      aria-hidden="true"
      class="inline-flex items-center justify-center bg-opacity-20 flex-shrink-0 w-5 h-5 rounded-lg text-success"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10 .5a9.5 9.5 0 1 0 9.5 9.5A9.51 9.51 0 0 0 10 .5Zm3.707 8.207-4 4a1 1 0 0 1-1.414 0l-2-2a1 1 0 0 1 1.414-1.414L9 10.586l3.293-3.293a1 1 0 0 1 1.414 1.414Z"
      />
    </svg>
    <p
      class="text-sm text-primary-200 max-w-[220px] ml-2 mr-3"
    >
      Item moved successfully.
    </p>
    <button
      class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-small gap-2 rounded-medium !gap-0 transition-transform-colors-opacity motion-reduce:transition-none w-10 h-10 data-[hover=true]:opacity-hover border-transparent font-semibold text-primary-200 bg-none min-w-5 p-0"
      type="button"
    >
      <svg
        class="w-3 h-3"
        fill="none"
        viewBox="0 0 14 14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 1l6 6m0 0l6 6M7 7l6-6M7 7l-6 6"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </button>
  </div>
</div>
`;
